Update classic settings tabs - follow-up PR

moon0326 opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

<!-- For bug fixes: If known, please provide links to help with traceability and escape analysis. -->
<!-- Please include a link to the issue of the bug being fixed, if one doesn't exist please create it. -->
<!-- If the PR that introduced the bug is known, please also add its link below. -->

This is a follow-up PR for #55912 to fix a bug and remove settings-content DIV.

1. Removed "div id='settings-content'" to maintain backward compatibility.
2. Adjusted styles to ensure compatibility with the storewide notice.
3. Adjusted CSS applied in https://github.com/woocommerce/woocommerce/pull/56412 to work with the new changes.


### Screenshots or screen recordings:

<!-- If this PR includes UI changes, please provide screenshots or a screen recording for clarity. -->
<!-- This section can be removed if not applicable. -->

| Before | After |
| ------ | ----- |
|  ![Screenshot 2025-03-18 at 1 20 14 PM](https://github.com/user-attachments/assets/55137848-114b-47f5-96e6-7592e3b3dcb9) |  ![Screenshot 2025-03-18 at 1 24 34 PM](https://github.com/user-attachments/assets/5d14106e-98d5-43c2-b7f1-aa123abf64ab)   |

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with this branch.
4. Navigate to WooCommerce -> Settings
5. Confirm style changes. 

If you don't see the notice, perform the following steps.

1. Install and activate `SQL Executioner`
2. Navigate to Tools -> SQL Executioner
3. Run the following query.

```
UPDATE wp_wc_admin_notes 
SET type = 'update', status='unactioned'
WHERE note_id IN (
    SELECT note_id FROM (
        SELECT note_id FROM wp_wc_admin_notes WHERE type='info' LIMIT 1
    ) AS subquery
);
```

<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

Update classic setting page styles to work with storewide notice.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

Commit:
Remove settings-content div

Commit:
Remove settings-content

Commit:
Update nox css settings

Commit:
Adjust store alert margin

Commit:
Add changefile(s) from automation for the following project(s): woocommerce, woocommerce/client/admin

Commit:
Merge branch 'trunk' into update/classic-settings-issue-with-notice

Commit:
Lint fixes

ilyasfoo reviewed (approved):
Tests well, code change LGTM. This isn't related to the PR, but might be productive to fix together anyway. I found the border color to not follow the theme set, I believe it should be `var(--wp-admin-theme-color)`:

<img width="400" alt="image" src="https://github.com/user-attachments/assets/d97dcaa6-3701-4dce-a24a-c3d27192c9ad" />

#### Review checklist

| Parameter  | Result |
| ------------- | ------------- |
| Approach  | ✅ All good |
| Implementation  | ✅ All good |
| Automated tests  | Not applicable |
| Code comments  | ✅ All good |
| Changelog  | ✅ All good |
| Documentation  | Not applicable |
| PR description  | ✅ All good |
| Addresses issue criteria  | ✅ All good |

#### Scenarios tested

| Scenario  | Result | Screenshot |
| ------------- | ------------- | ------------- |
| Settings header background when a note is showing | ✅ Functionality worked correctly | <img width="730" alt="image" src="https://github.com/user-attachments/assets/91c6eff1-ff25-4761-8751-cfe2efd30902" /> |

Commit:
Merge branch 'trunk' into update/classic-settings-issue-with-notice

Commit:
Update active tab border color and add tab hover effect

moon0326 commented:
> Tests well, code change LGTM. This isn't related to the PR, but might be productive to fix together anyway. I found the border color to not follow the theme set, I believe it should be `var(--wp-admin-theme-color)`:

Good catch! I've updated it in [Update active tab border color and add tab hover effect](https://github.com/woocommerce/woocommerce/pull/56502/commits/f4e8df3d8a2888c9854d7b38bfcbca9c4b59050c). I also added hover effect as suggested by Chi from pdibGW-44C-p2#comment-3114

Commit:
Merge branch 'trunk' into update/classic-settings-issue-with-notice

Commit:
Merge branch 'trunk' into update/classic-settings-issue-with-notice

Commit:
Merge branch 'trunk' into update/classic-settings-issue-with-notice

rjchow commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -8923,9 +8938,13 @@ body.woocommerce_page_wc-settings {
 			color: #070707;
 		}
+
+		.nav-tab:hover:not(.nav-tab-active) {
```
Let's add focus to this too, I realised that it's important for a11y reasons

moon0326 replied:
Thanks @rjchow 
Fixed it in [Fix hover and focus color](https://github.com/woocommerce/woocommerce/pull/56502/commits/f876696b9046b2b774c09f46672a9843fd66ea75)

chihsuan reviewed (commented):
Thanks @moon0326 Overall, changes look good! Just one thing, I noticed that the text color on the active tab hover remains black instead of using the theme color. It should be consistent with the extensions page behavior.

https://github.com/user-attachments/assets/e7405028-2f7d-4b94-9cbe-ac0a99454596





___

#### Test environment

<details>
WordPress: 6.4
Theme: Twenty Twenty-Four Version: 1.0
WooCommerce: Development version with update/classic-settings-issue-with-notice branch
PHP: 8.2
Web Server: Nginx
Browser: Chrome
OS: macOS Sonoma 14.3.1
</details>

#### Review checklist

| Parameter | Result |
| ------------- | ------------- |
| Approach | ✅ All good |
| Implementation | ✅ All good |
| Automated tests | Not applicable |
| Code comments | ✅ All good |
| Changelog | ✅ All good |
| Documentation | Not applicable |
| PR description | ✅ All good |
| Addresses issue criteria | ✅ All good |

#### Scenarios tested

| Scenario | Result | Screenshot |
| ------------- | ------------- | ------------- |
| Settings header background when a notice is showing | ✅ Functionality worked correctly | 
<img width="1147" alt="Screenshot 2025-03-20 at 14 34 47" src="https://github.com/user-attachments/assets/d9e688b8-b354-41bb-bb54-24d5bbe5f0a8" /> |
| Active tab border color | ✅ Fixed with theme color | Updated in commit f4e8df3 |
| Tab hover effect | one minor issue | <img width="686" alt="Screenshot 2025-03-20 at 14 31 52" src="https://github.com/user-attachments/assets/e189da2b-4f17-43df-bb1c-8f0aaa343994" />  |

rjchow reviewed (approved):
Tested well according to instructions! There's just one small comment I made for accessibility reasons, but otherwise pre-approving

Commit:
Fix hover and focus color

moon0326 commented:
> Thanks @moon0326 Overall, changes look good! Just one thing, I noticed that the text color on the active tab hover remains black instead of using the theme color. It should be consistent with the extensions page behavior.

Thank you! Fixed it in [Fix hover and focus color](https://github.com/woocommerce/woocommerce/pull/56502/commits/f876696b9046b2b774c09f46672a9843fd66ea75)

Commit:
Merge branch 'trunk' into update/classic-settings-issue-with-notice
